finrl/env/env_stocktrading_cashpenalty.py

```diff
@@ -52,28 +56,38 @@ class StockTradingEnvCashpenalty(gym.Env):
     def __init__(
         self,
         df,
-        transaction_cost_pct=3e-3,
+        buy_cost_pct=3e-3,
+        sell_cost_pct=3e-3,
         date_col_name="date",
         hmax=10,
+        min_shares=1,
         turbulence_threshold=None,
         print_verbosity=10,
         initial_amount=1e6,
         daily_information_cols=["open", "close", "high", "low", "volume"],
         cache_indicator_data=True,
         cash_penalty_proportion=0.1,
+        negative_asset_penalty=0,
         random_start=True,
+        discrete_actions=False,
+        currency="$",
     ):
         self.df = df
         self.stock_col = "tic"
         self.assets = df[self.stock_col].unique()
         self.dates = df[date_col_name].sort_values().unique()
         self.random_start = random_start
+        self.discrete_actions = discrete_actions
+        self.currency = currency
 
-        self.df = self.df.set_index(date_col_name)
-        self.hmax = hmax
-        self.initial_amount = initial_amount
+        self.df = self.df.set_index(date_col_name
+        self.min_shares = min_shares
+        self.hmax = hmax / min_shares
+        self.initial_amount = initial_amount / min_shares
```

Comment:
what is the purpose of min_shares?

Reply:
In some countries, including Indonesia, we can only buy with a minimum of 100 shares. So this acts as a scaling factor

Reply:
Ok, in that case, I don't think that scaling initial_amt by min_shares is correct.

Reply:
Hmm, could you please elaborate more on how should we tackle this problem in a simpler solution rather than scaling by min_shares?

Reply:
ok. Initial amt is the amount of cash you start with. even if you can trade only in increments of 100 shares, it should not change the initial amount of cash but only the numbers of shares traded at each step.

Reply:
Yes, I agree with you. However, to simplify the work, I think we can just scale down the `initial_amount` and `hmax` by `min_shares`, and think that during each trade, when we buy 1 share is equal to what we buy `min_shares` of shares in real life

Reply:
I don't agree with this. 
The cash should not be impacted by the min shares. It's ok if this is a requirement for the actions space, but a user putting in a initial_cash amount would be surprised to see that their cash on hand had been cut by the factor `min_shares` at the first step and be very confused.

Reply:
ooh, the account balance is a purse, managing only "cash in" and "cash out", right?  
We may consider to accommodate several cases: 1. stock shares in integer;  2. stock shares in unit of "100 shares";  3. btc allows fractional shares;  4. Robinhood may allow cash and make it into fractional shares.  Or other cases that I am not aware of.

Reply:
@XiaoYangLiu-FinRL Yes, I agree we have to accommodate all of those possibilities.

@spencerR1992 That's a valid point. That's why in the logging I multiply it back with the `min_cash` parameter. 

```
logger.record("environment/total_assets", int(self.min_shares*self.account_information['total_assets'][-1]))
```

CMIIW, I think if we only scaled-down the `hmax` without also scaling down the `initial_amount` then it will be not fair...